packages/extension-dapp/src/compat/web3Source.ts

```diff
@@ -0,0 +1,137 @@
+//   };
+// }
+
+// interface SingleSourceAccount {
+//   address: string;
+//   assets: { assetId: number }[];
+//   name: string;
+// }
+
+// interface SingleSource {
+//   accounts$: Subscriber<SingleSourceAccount[]>;
+//   environment$: string[];
+//   signer: Signer;
+// }
+
+// interface SingleWindow extends InjectedWindow {
+//   SingleSource: SingleSource;
+// }
+// interface SingleSourceAccount {
+//     address: string;
+//     assets: { assetId: number }[];
+//     name: string;
+//   }
+
+//   interface SingleSource {
+//     accounts$: Subscriber<SingleSourceAccount[]>;
+//     environment$: string[];
+//     signer: Signer;
+//   }
+
+interface Web3Window extends InjectedWindow {
+  web3: Web3;
+  // this is injected by metaMask
+  ethereum: any;
+}
+
+// transfor the Web3 accounts into a simple address/name array
+function transformAccounts (accounts: string[]): InjectedAccount[] {
+  return accounts.map((acc, i) => {
+    return { address: acc, name: 'MetaMask Address #' + i.toString() };
+  });
+}
+
+// add a compat interface of SingleSource to window.injectedWeb3
+function injectWeb3 (win: Web3Window): void {
+  // let accounts: InjectedAccount[] = [];
+
+  // we don't yet have an accounts subscribe on the interface, simply get the
+  // accounts and store them, any get will resolve the last found values
+  //   win.web3.accounts$.subscribe((_accounts): void => {
+  //     accounts = transformAccounts(_accounts);
+  //   });
+
+  // decorate the compat interface
+  win.injectedWeb3.Web3Source = {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars,@typescript-eslint/require-await
+    enable: async (_: string): Promise<Injected> => {
+      win.web3 = new Web3(win.ethereum);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+      await win.ethereum.enable();
+
+      // let mainAccount=(await win.web3.eth.getAccounts())[0]
+      return {
+        accounts: {
+          get: async (): Promise<InjectedAccount[]> => {
+            console.log('fetching accounts');
+            console.log(await win.web3.eth.getAccounts());
+
+            return transformAccounts(await win.web3.eth.getAccounts());
+          },
+          subscribe: (cb: (accounts: InjectedAccount[]) => void): (() => void) => {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+            const sub = win.ethereum.on('accountsChanged', function (accounts:string[]) {
+              cb(transformAccounts(accounts));
+            });
+
+            return (): void => {
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+              sub.unsubscribe();
+            };
+          }
+        },
+        signer: {
+          signPayload: async (payload: SignerPayloadJSON) : Promise<SignerResult> => {
```

Comment:
@jacogr could you take a look at this implementation of signPayload pls. I'm not really used to polkadot transactions yet. What is the signer supposed to sign? The function is signing a stringified paylaod but it doesn't seem to work. Thanks in advance

Reply:
lol okay so you anticipated my question

Reply:
:D